### PR TITLE
Refactor rz_core_theme_list() to return RzPVector

### PR DESCRIPTION
**Pull request by p371jah**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/core/cmd/cmd_eval.c [hunk 1]
```diff
@@ -124,22 +91,20 @@ RZ_API char *rz_core_theme_get(RzCore *core) {
...
 
 /**
  * \brief      Returns the list of the rizin themes.
  *
  * \param      core  The RzCore struct to use
- * \return     On success, an RzList pointer, otherwise NULL.
+ * \return     On success, an RzPVector pointer, otherwise NULL.
  */
-RZ_API RZ_OWN RzList /*<char *>*/ *rz_core_theme_list(RZ_NONNULL RzCore *core) {
+RZ_API RZ_OWN RzPVector /*<char *>*/ *rz_core_theme_list(RZ_NONNULL RzCore *core) {
```
Comment (XVilka) [inline librz/core/cmd/cmd_eval.c]: The same question about the name. It's better to remove the list part. `rz_core_themes()`? Or `rz_core_theme_vec()`?

**Review by XVilka — CHANGES_REQUESTED** [review]
LGTM except the function name

File: librz/core/cmd/cmd_eval.c [hunk 2]
```diff
@@ -162,32 +127,46 @@ RZ_API RZ_OWN RzList /*<char *>*/ *rz_core_theme_list(RZ_NONNULL RzCore *core) {
...
+		if (strcmp(fn, core->curtheme)) {
+			continue;
+		}
+		if (mode == RZ_CONS_PAL_SEEK_PREVIOUS) {
+			if (idx == 0) {
+				break;
 			}
+			fn = rz_pvector_at(files, idx - 1);
+		} else if (mode == RZ_CONS_PAL_SEEK_NEXT) {
+			if (idx == rz_pvector_len(files) - 1) {
```
Comment (wargio) [inline librz/core/cmd/cmd_eval.c]:
```suggestion
			if (idx >= rz_pvector_len(files) - 1) {
```

File: librz/core/cmd/cmd_eval.c [hunk 3]
```diff
@@ -162,32 +127,46 @@ RZ_API RZ_OWN RzList /*<char *>*/ *rz_core_theme_list(RZ_NONNULL RzCore *core) {
...
+	void **iter;
+	size_t idx;
+	RzPVector *files = rz_core_theme_list(core);
+	rz_pvector_enumerate (files, iter, idx) {
+		const char *fn = *iter;
+		if (strcmp(fn, core->curtheme)) {
+			continue;
+		}
+		if (mode == RZ_CONS_PAL_SEEK_PREVIOUS) {
+			if (idx == 0) {
```
Comment (wargio) [inline librz/core/cmd/cmd_eval.c]:
```suggestion
			if (idx < 1) {
```

File: librz/core/cmd/cmd_eval.c [hunk 4]
```diff
@@ -162,32 +127,46 @@ RZ_API RZ_OWN RzList /*<char *>*/ *rz_core_theme_list(RZ_NONNULL RzCore *core) {
...
+
+	void **iter;
+	size_t idx;
+	RzPVector *files = rz_core_theme_list(core);
+	rz_pvector_enumerate (files, iter, idx) {
+		const char *fn = *iter;
+		if (strcmp(fn, core->curtheme)) {
+			continue;
+		}
+		if (mode == RZ_CONS_PAL_SEEK_PREVIOUS) {
```
Comment (wargio) [inline librz/core/cmd/cmd_eval.c]: maybe use a switch case with the default?

File: librz/core/cmd/cmd_eval.c (hunk 2 above)
Comment (p371jah) [reply librz/core/cmd/cmd_eval.c]: `rz_pvector_enumerate` iterates from `0` to `rz_pvector_len(pvec) - 1`

**Review by XVilka — APPROVED** [review]

**Pull request merged**